### PR TITLE
fix: auth token multiple misleading source

### LINKED ISSUE #271: Clarity about the source of auth token used
Right now, we have three ways of finding an auth token:

- In settings.json, from `turso auth login`
- In TURSO_API_TOKEN env var
- In `-t` CLI flag

Since all three may be present, it's potentially confusing since the CLI is not very clear about which one of these it's using when a command runs successfully.  Also, if the CLI fails because the chosen token is expired or invalid, there is no feedback about which one was problematic, leaving the user to potentially take action that doesn't resolve the issue.

Proposal:

- Let's define the order in which these sources are used. I suggest: 1. CLI flag, 2. env var, 3. settings.json
- For all (non-auth) commands that require a token, always look at all sources. If more than one token is found, immediately print a warning message listing all sources where tokens were found, and which one is being used to complete the command.

### PR DESCRIPTION
**Pull request by StefanoSaffran**
This PR aims to warn the user about the source of the token

closes #271

### CODE REVIEW

**Comment by athoscouto** [comment]
The first two commits look good!

I'm not totally comfortable with the third one though.
I think the env should have priority over the settings file.
But if somebody has an env set and the token is valid, I don't think we should warn them about it.

If the token on the env is invalid, we can remember that the env is set.
If they try to login or logout, we should also warn them that this will not work as expected, since they have an env set.

I'm repeating myself since I already said those things on the issue.

But, once again, I see two downsides with those blanket warnings:
- It messes up scripts. People expect a consistent output from a command, if you add unexpected things to `stdout` it can break scripts that use our CLI. 
- It gets annoying, you're almost convincing them to not use the token env. I don't think it is good DX to have to check your settings file (God knows where it is) to remove the token before you add it to your env var.

File: internal/cmd/auth.go [hunk 1]
```diff
@@ -181,7 +179,7 @@ func auth(cmd *cobra.Command, args []string, path string) error {
 		fmt.Printf("✔  Success! Logged in as %s\n", username)
 
 		firstTime := settings.RegisterUse("auth_login")
-		client, err := createTursoClient()
+		client, err := createTursoClientFromAccessToken(false)
```
Comment (haaawk) [inline internal/cmd/auth.go:182]: This commit does not compile. createTursoClientFromAccessToken is introduced in later commit.

File: internal/cmd/db.go [hunk 2]
```diff
@@ -121,22 +121,25 @@ var dbCmd = &cobra.Command{
...
 	settings, err := settings.ReadSettings()
 	if err != nil {
 		return "", fmt.Errorf("could not read local settings")
 	}
+	settingsToken := settings.GetToken()
 
-	token := settings.GetToken()
-	if !isJwtTokenValid(token) {
+	if envToken != "" && settingsToken != "" && warnMultipleAccessTokenSources {
+		fmt.Println("Warning: found both environment and configuration token. Proceeding with environment one")
```
Comment (haaawk) [inline internal/cmd/db.go:133]: I think we could improve the the error message to be more user friendly. Something like:
```
User logged in as <fetch user name from settings> but TURSO_API_TOKEN environment variable set so using it instead.
```

File: internal/cmd/db_passwd.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ var changePasswordCmd = &cobra.Command{
 
 		bar := prompt.Spinner("Changing password...")
 		defer bar.Stop()
-		client, err = createTursoClient()
+		client, err = createTursoClientFromAccessToken(false)
```
Comment (haaawk) [inline internal/cmd/db_passwd.go:57]: I'm wondering why we can't reuse the client from line 29

**Comment by haaawk** [comment]
LGTM in general. @CodingDoug do you have any opinion regarding removing the ability to specify access token using command-line flag?

**Comment by CodingDoug** [comment]
> LGTM in general. @CodingDoug do you have any opinion regarding removing the ability to specify access token using command-line flag?

@haaawk I re-read my comments in the original issue and found that I couldn't come up with a strong use case for allowing the command line flag.  I suppose I still agree with my past self. :)

**Comment by haaawk** [comment]
> > LGTM in general. @CodingDoug do you have any opinion regarding removing the ability to specify access token using command-line flag?
> 
> @haaawk I re-read my comments in the original issue and found that I couldn't come up with a strong use case for allowing the command line flag. I suppose I still agree with my past self. :)

Good - let's remove it then

File: internal/cmd/db_passwd.go (hunk 3 above)
Comment (StefanoSaffran) [reply internal/cmd/db_passwd.go:57]: Maybe the idea when it was created was to handle the error that wasn't handled in the first one.
https://github.com/chiselstrike/turso-cli/commit/adde529ea5e9e35ca7bbbc53f8c8f784112fe446

I tested it now removing the second client and it works.
I think it is safe to remove it, right?

Comment (haaawk) [reply internal/cmd/db_passwd.go]: if the test passes then it should be fine.

**Review by haaawk — APPROVED** [review]

**Pull request merged**